pageserver/src/repository.rs

```diff
@@ -168,6 +150,29 @@ pub trait Timeline: Send + Sync {
     fn get_current_logical_size_non_incremental(&self, lsn: Lsn) -> Result<usize>;
 }
 
+pub trait TimelineWriter: std::ops::Deref<Target = dyn Timeline> {
```

Comment:
Questionable if Deref is the right tool here. I only used it because I didn't want to modify even more existing code.

Reply:
The idea is that you can call the Timeline::get_* functions directly on the writer? I.e. so that `writer.get_page_at_lsn(...)` works? Seems reasonable to me, but needs comments.